src/main/java/mate/academy/bookshop/validation/FieldMatchValidator.java

```diff
@@ -24,16 +23,12 @@ public boolean isValid(Object object, ConstraintValidatorContext constraintValid
     }
 
     private Object getFieldValueByName(Object object, String fieldName) {
-        Field[] fields = object.getClass().getDeclaredFields();
-        Field field = Arrays.stream(fields)
-                .filter(f -> f.getName().equals(fieldName))
-                .findFirst()
-                .orElseThrow(() -> new IllegalArgumentException("Field with this name not found"));
-        field.setAccessible(true);
         try {
+            Field field = object.getClass().getDeclaredField(fieldName);
+            field.setAccessible(true);
             return field.get(object);
-        } catch (IllegalAccessException e) {
-            throw new RuntimeException("Can't get field value");
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get field value", e);

```

Comment:
as like the argument of field name that we received is invalid
```suggestion
            throw new IllegalArgumentException("Can't get field value", e);
```